addons/hr_attendance/models/res_company.py

```diff
@@ -36,6 +39,29 @@ def _compute_attendance_kiosk_url(self):
         for company in self:
             company.attendance_kiosk_url = url_join(company.get_base_url(), '/hr_attendance/%s' % company.attendance_kiosk_key)
 
+    # ---------------------------------------------------------
+    # ORM Overrides
+    # ---------------------------------------------------------
+    def _init_column(self, column_name):
+        """ Initialize the value of the given column for existing rows.
+            Overridden here because we need to generate different access tokens
+            and by default _init_column calls the default method once and applies
+            it for every record.
+        """
+        if column_name != 'attendance_kiosk_key':
+            super(ResCompany, self)._init_column(column_name)
+        else:
+            self.env.cr.execute("SELECT id FROM %s WHERE attendance_kiosk_key IS NULL" % self._table)
+            acc_ids = self.env.cr.dictfetchall()
+            values_args = [(acc_id['id'], self._default_employee_token()) for acc_id in acc_ids]
```

Comment:
Blind copy pasting is never a good idea. 
what does acc_ids stand for here?

```suggestion
            attendance_ids = self.env.cr.dictfetchall()
            values_args = [(attendance_id['id'], self._default_company_token()) for attendance_id in attendance_ids]
```